extensions/amp-auto-ads/0.1/ad-network-config.js

```diff
@@ -123,3 +127,73 @@ class AdSenseNetworkConfig {
     };
   }
 }
+
+
+/**
+ * @implements {AdNetworkConfigDef}
+ */
+class DoubleclickNetworkConfig {
+  /**
+   * @param {!Element} autoAmpAdsElement
+   */
+  constructor(autoAmpAdsElement) {
+    this.autoAmpAdsElement_ = autoAmpAdsElement;
+  }
+
+  /**
+   * @param {!Window} win
+   */
+  isEnabled(win) {
+    const branch = getAdSenseAmpAutoAdsExpBranch(win);
+    return branch != AdSenseAmpAutoAdsHoldoutBranches.CONTROL;
+  }
+
+  /** @override */
+  getConfigUrl() {
+    const docInfo = Services.documentInfoForDoc(this.autoAmpAdsElement_);
+    const canonicalHostname = parseUrlDeprecated(docInfo.canonicalUrl).hostname;
+    return buildUrl('//pagead2.googlesyndication.com/getconfig/ama', {
+      'client': this.autoAmpAdsElement_.getAttribute('data-ad-legacy-client'),
+      'plah': canonicalHostname,
+      'ama_t': 'amp',
+      'url': docInfo.canonicalUrl,
+    }, 4096);
+  }
+
+  /** @override */
+  getAttributes() {
+    const attributes = dict({
+      'type': 'doubleclick',
+      'data-slot': this.autoAmpAdsElement_.getAttribute('data-slot'),
+      'json': this.autoAmpAdsElement_.getAttribute('data-json'),
+    });
+    const experimentJson = tryParseJson(
+        this.autoAmpAdsElement_.getAttribute('data-experiment'));
+    if (experimentJson) {
+      if (experimentJson['height']) {
```

Comment:
I think this is a bit of a misuse of the attributes and I don't really like the way this then has to be cleaned up in placement.js before creating the ad element.

I think it would be better to create a new typedef that specifies parameters for the ad placements, such as height and width. The AdNetworkConfigDef interface can then have a getPlacementParameters method which returns this. placeAd would then take this as a parameter and read from it what it needs.